addons/mail/models/discuss/ir_websocket.py

```diff
@@ -99,8 +203,9 @@ def _update_bus_presence(self, inactivity_period, im_status_ids_by_model):
     def _on_websocket_closed(self, cookies):
         super()._on_websocket_closed(cookies)
         if self.env.user and not self.env.user._is_public():
+            self.env["mail.presence"].search([("user_id", "=", self.env.uid)]).status = "offline"
```

Comment:
I think it would be easier to build the domain conditionally rather than assigning the status to the result, since both branches perform the same operation on it.

Reply:
same comment about one of the branches using `sudo`